From: Peter Kokot <peterkokot@gmail.com>
Subject: Enhance php-config script

Changes:

* Add PHP API version number to php-config

  This adds a new option '--php-api' to the php-config script to output
  the PHP API version number as defined in the main/php.h file
  (PHP_API_VERSION). There is also ZEND_MODULE_API_NO that matches the
  PHP API value and is used in the extension directory path. Perhaps,
  the --php-api should output the ZEND_MODULE_API_NO, or another option
  added like --zend-api. There are also PHP installations, that already
  use similar patched approaches, such as --phpapi on Debian and Ubuntu,
  where the value matches the ZEND_MODULE_API_NO.

* Remove sed from php-config script

  During the configuration step, the Autoconf configure options are
  replaced in main/build-defs.h.in and scripts/php-config.in. Instead of
  doing redundant quotes removal step each time the php-config script is
  run by the user on installed PHP, they can be removed once before the
  variable substitution step directly at the PHP configuration phase
  when building PHP. This also removes the unused CONFIGURE_OPTIONS
  Make variable in root and phpize generated Makefiles. The configure
  options in main/build-defs.h.in remain quoted as before.

  Also, sed substituted value can have issues in cases where build is
  happening on one machine and targeted machine has different path to
  sed command. For example, on build machine it is in /usr/bin/sed, and
  on target in /bin/sed.

  This should be further improved because some PHP installations
  manually adjust the php-config script and inject variables in
  php-config script, such as CFLAGS. For example, variables passed to
  configure after quotes removal can include malformed value with spaces
  and without any quotes around them. Perhaps removing quotes should be
  done only on configure options and not variable arguments.
---
 build/php.m4                 | 4 +++-
 configure.ac                 | 3 +++
 scripts/man1/php-config.1.in | 4 ++++
 scripts/php-config.in        | 8 ++++----
 4 files changed, 14 insertions(+), 5 deletions(-)

diff --git a/build/php.m4 b/build/php.m4
index 933ada51dc..4909484a3b 100644
--- a/build/php.m4
+++ b/build/php.m4
@@ -2159,7 +2159,9 @@ EOF
   chmod +x $1
   CONFIGURE_COMMAND="$CONFIGURE_COMMAND $CONFIGURE_OPTIONS"
   PHP_SUBST_OLD(CONFIGURE_COMMAND)
-  PHP_SUBST_OLD(CONFIGURE_OPTIONS)
+  dnl Remove quotes.
+  CONFIGURE_OPTIONS=$(echo $CONFIGURE_OPTIONS | $SED -e "s#'##g")
+  AC_SUBST([CONFIGURE_OPTIONS])
 ])
 
 dnl
diff --git a/configure.ac b/configure.ac
index 01e2e7ab79..bf80beb052 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1482,6 +1482,9 @@ AC_SUBST(EXPANDED_LOCALSTATEDIR)
 AC_SUBST(EXPANDED_PHP_CONFIG_FILE_PATH)
 AC_SUBST(EXPANDED_PHP_CONFIG_FILE_SCAN_DIR)
 
+PHP_API_VERSION=$($EGREP '#define PHP_API_VERSION ' $srcdir/main/php.h | "${SED}" -e 's/^[^0-9]*//')
+AC_SUBST([PHP_API_VERSION])
+
 PHP_UTILIZE_RPATHS
 
 PHP_REMOVE_USR_LIB(PHP_LDFLAGS)
diff --git a/scripts/man1/php-config.1.in b/scripts/man1/php-config.1.in
index 8ccd171641..333978f35c 100644
--- a/scripts/man1/php-config.1.in
+++ b/scripts/man1/php-config.1.in
@@ -39,6 +39,10 @@ Directory where extensions are searched by default
 Directory prefix where header files are installed by default
 .TP
 .PD 0
+.B \-\-php-api
+PHP API version number
+.TP
+.PD 0
 .B \-\-php-binary
 Full path to php CLI or CGI binary
 .TP
diff --git a/scripts/php-config.in b/scripts/php-config.in
index 9271e87286..0ab01a82cb 100644
--- a/scripts/php-config.in
+++ b/scripts/php-config.in
@@ -1,6 +1,5 @@
 #! /bin/sh
 
-SED="@SED@"
 prefix="@prefix@"
 datarootdir="@datarootdir@"
 exec_prefix="@exec_prefix@"
@@ -21,6 +20,7 @@ configure_options="@CONFIGURE_OPTIONS@"
 php_sapis="@PHP_INSTALLED_SAPIS@"
 ini_dir="@EXPANDED_PHP_CONFIG_FILE_SCAN_DIR@"
 ini_path="@EXPANDED_PHP_CONFIG_FILE_PATH@"
+php_api="@PHP_API_VERSION@"
 
 # Set php_cli_binary and php_cgi_binary if available
 for sapi in $php_sapis; do
@@ -41,9 +41,6 @@ else
   php_binary="$php_cgi_binary"
 fi
 
-# Remove quotes
-configure_options=`echo $configure_options | $SED -e "s#'##g"`
-
 case "$1" in
 --prefix)
   echo $prefix;;
@@ -57,6 +54,8 @@ case "$1" in
   echo $extension_dir;;
 --include-dir)
   echo $include_dir;;
+--php-api)
+  echo $php_api;;
 --php-binary)
   echo $php_binary;;
 --php-sapis)
@@ -84,6 +83,7 @@ Options:
   --extension-dir     [$extension_dir]
   --include-dir       [$include_dir]
   --man-dir           [$man_dir]
+  --php-api           [$php_api]
   --php-binary        [$php_binary]
   --php-sapis         [$php_sapis]
   --ini-path          [$ini_path]
